server/controllers/profile.js

```diff
@@ -0,0 +1,100 @@
+const mongoose = require('mongoose');
+
+const Profile = require('../models/Profile');
+
+const asyncHandler = require("express-async-handler");
+
+// @route POST 
+// Create New User Profile
+exports.newUser = asyncHandler(async (req, res) => {
+    const { userId, firstName, lastName, description, location, isDogOwner, isDogSitter, rating, hourlyRate, tagLine } = req.body;
+    
+    const profile = await Profile.create({
+        userId,
+        firstName,
+        lastName,
+        description,
+        location,
+        isDogOwner,
+        isDogSitter,
+        rating,
+        hourlyRate,
+        tagLine
+    });
+
+    if (profile) {
+         res.status(201).json({
+            userId: profile.userId,
+            firstName: profile.FirstName,
+            lastName: profile.lastName,
+            description: profile.description,
+            location: profile.location,
+            isDogSitter: profile.isDogSitter,
+            rating: profile.rating,
+            hourlyRate: profile.hourlyRate,
+            tagLine: profile.tagLine,
+        })
+    } else {
+       res.status(400).json({ message: "Invalid User Data"})
+    }
+});
+
+// //@route UPDATE
+// //update profiles 
+exports.editProfile = asyncHandler(async (req, res) => {
+    const userId = req.params.id
+    
+    try {
+        const updateProfile = await Profile.findOne({userId: userId}, function(profile, error) {
+        if(error) { res.status(400).json({ message: "Invalid User Data"}) }
+        else {
+            for (const params of Object.params(req.body)) {
+                profile[params] = req.body[params]
+            }
+            profile.save(res.status(200).json({profile: updateProfile}))
+        }

```

Comment:
This could be cleaned up with [findOneAndUpdate](https://mongoosejs.com/docs/populate.html), also the callback here is not necessary as findOne is passes back a Promise which we can await.

Reply:
Made the requested changes, let me know how it looks.